sqlitecluster/SQLite.cpp

```diff
@@ -665,6 +641,35 @@ bool SQLite::_writeIdempotent(const string& query, SQResult& result, bool always
 bool SQLite::prepare(uint64_t* transactionID, string* transactionhash) {
     SASSERT(_insideTransaction);
 
+    // Pick a journal for this transaction.
+    const int64_t journalID = _sharedData.nextJournalCount++;
+    _journalName = _journalNames[journalID % _journalNames.size()];
+
+    // It's possible to attempt to commit a transaction with no writes. We'll skip truncating the journal in this case to avoid
+    // Turning a no-op into a write.
+    if (_uncommittedQuery.size()) {
```

Comment:
Just so we aren't skipping over journals for noops, can we increment nextJournalCount inside this block instead? It's minor but avoids one tiny atomic overhead for things we won't even use it for.

Reply:
Unless somehow the nextJournalCount is necessary to increment even for noops...

Reply:
I'm going to simplify this a different way and just always do the delete rather than special casing this, as even though it's allowed, we don't expect it to happen.